### PR TITLE
feat(text-input-group): added component

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/4374

### CODE REVIEW

**Review by mcarrano — COMMENTED** [review]
This looks good @mcoker . I just have a question about what the responsive behavior should be. Looks like right now it just clips when I shrink the screen. Expect some type of truncation. Will React need to handle that?

**Comment by mcoker** [comment]
@mcarrano in order for the text to truncate with CSS, text needs to overflow the container. In this case, the chip group is overflowing the container, so the text in the chips doesn't truncate. We could try some other sort of effect.

**Comment by mcarrano** [comment]
So I guess that the Chip group needs to handle the overflow then?

**Review by nicolethoen — APPROVED** [review]
I think the basic html structure makes sense and should be fairly straight forward to bring into react. Responsive behaviors are a good question. But if we can put breakpoint mods on various sub components inside a text-input-group, then it may work out no problem

File: src/patternfly/components/TextInputGroup/examples/TextInputGroup.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    {{/text-input-group-main}}
+  {{/text-input-group-row}}
+{{/text-input-group}}
+```
+
+### Utilities and icon
+```hbs
+{{#> text-input-group text-input-group--modifier="pf-m-icon"}}
+  {{#> text-input-group-row}}
+      {{#> text-input-group-main}}
```
Comment (mattnolting) [inline src/patternfly/components/TextInputGroup/examples/TextInputGroup.md:26]:
```suggestion
    {{#> text-input-group-main}}
```

File: src/patternfly/components/TextInputGroup/examples/TextInputGroup.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      {{#> text-input-group-text}}
+        {{> text-input-group-icon}}
```
Comment (mattnolting) [inline src/patternfly/components/TextInputGroup/examples/TextInputGroup.md:28]: Looks like this partial is missing.

**Review by mattnolting — COMMENTED** [review]
This looks great. I do wonder if you need `.pf-c-text-input-group__row` or those styles can be applied directly to `.pf-c-text-input-group`.

Looks like your

**Comment by mcoker** [comment]
> I do wonder if you need .pf-c-text-input-group__row or those styles can be applied directly to .pf-c-text-input-group.

@mattnolting it's modeled off of the existing search input, and the row class is there to separate the search input stuff from the menu that popped out of the bottom of the component. I don't think it's necessary here, though I left it since I imagine that container could be useful or allow it to be styled more flexibly. Though since it's beta, we can just add it later if we end up needing it.

Updated that and fixed the missing partial.

**Review by mattnolting — APPROVED** [review]
LPTM 🚀  Ship it!!

**Pull request merged**